src/framework/uicomponents/view/buttonboxmodel.h

```diff
@@ -0,0 +1,188 @@
+        Ok                = 0x0000001,
+        Continue          = 0x0000002,
+        RestoreDefaults   = 0x0000004,
+        Reset             = 0x0000008,
+        Apply             = 0x0000010,
+        Help              = 0x0000020,
+        Discard           = 0x0000040,
+        Cancel            = 0x0000080,
+        Close             = 0x0000100,
+        Ignore            = 0x0000200,
+        Retry             = 0x0000400,
+        Abort             = 0x0000800,
+        NoToAll           = 0x0001000,
+        No                = 0x0002000,
+        YesToAll          = 0x0004000,
+        Yes               = 0x0008000,
+        Open              = 0x0010000,
+        DontSave          = 0x0020000,
+        SaveAll           = 0x0040000,
+        Save              = 0x0080000,
+        Next              = 0x0100000,
+        Back              = 0x0200000,
+        Select            = 0x0400000,
+        Clear             = 0x0800000,
+        Done              = 0x1000000,
+
+        CustomButton      = 0x2000000,
+
+        FirstButton       = Ok,
+        LastButton        = CustomButton
+    };
+
+    Q_ENUM(ButtonType)
+
+    enum ButtonRole { // Keep updated with buttonRoleLayouts
+        AcceptRole,
+        RejectRole,
+        DestructiveRole,
+        ResetRole,
+        ApplyRole,
+        RetryRole,
+        HelpRole,
+        ContinueRole,
+        BackRole,
+        CustomRole
+    };
+
+    Q_ENUM(ButtonRole)
+
+    QVariant data(const QModelIndex& index, int role) const override;
+    int rowCount(const QModelIndex& parent = QModelIndex()) const override;
+    QHash<int, QByteArray> roleNames() const override;
+
+    int buttons() const;
+
+    Q_INVOKABLE void load();
+    Q_INVOKABLE void addCustomButton(int index, QString text, int role, bool isAccent, bool isLeftSide, QString navigationName);
+
+public slots:
+    void setButtons(const int& buttons);
+
+signals:
+    void buttonsChanged(const int& buttons);
+
+private:
+    enum Roles {
+        ButtonText = Qt::UserRole + 1,
+        Type,
+        Accent,
+        IsLeftSide,
+        CustomButtonIndex,
+        NavigationName,
+        NavigationColumn
+    };
+
+    const std::vector<ButtonRole> chooseButtonLayoutType();
+
+    struct LayoutButton {
+        QString text;
+        ButtonType buttonType;
+        ButtonRole buttonRole;
+        bool isAccent;
+        bool isLeftSide = false;
+        int customButtonIndex = -1;
+        QString navigationName = "";
+        int navigationColumn = 0;
+
+        LayoutButton(QString _text, ButtonType _buttonType,  ButtonRole _buttonRole, bool _isAccent, bool _isLeftSide = false,
+                     QString _navigationName = "")
+            : text(_text), buttonType(_buttonType),  buttonRole(_buttonRole), isAccent(_isAccent), isLeftSide(_isLeftSide),
+            navigationName(_navigationName)
+        {
+        }
+    };
+
+    const char* c = "uicomponents"; //To make Following Hash look neat
```

Comment:
This doesn't work: the context must be written inline, otherwise `lupdate` won't recognise the translatable string. Unfortunately.

(I'm currently collecting all PRs that affects translatable strings, so that's why I'm randomly posting a random comment 😁)